### PR TITLE
Update aliases event after deletion

### PR DESCRIPTION
**Pull request by erikjohnston**
Attempt to update the appropriate `m.room.aliases` event after deleting an alias. This may fail due to the deleter not being in the room.

Will also check if the canonical alias of the event is set to the deleted alias, and if so will attempt to delete it.

### CODE REVIEW

**Comment by erikjohnston** [comment]
Sytest: https://github.com/matrix-org/sytest/pull/216

**Comment by NegativeMjark** [comment]
LGTM

File: synapse/handlers/directory.py [hunk 1]
```diff
@@ -93,7 +95,7 @@ def create_appservice_association(self, service, room_alias, room_id,
         yield self._create_association(room_alias, room_id, servers)
 
     @defer.inlineCallbacks
-    def delete_association(self, user_id, room_alias):
+    def delete_association(self, requester, user_id, room_alias):
```
Comment (NegativeMjark) [inline synapse/handlers/directory.py]: Is the user_id different from the requester?

Comment (erikjohnston) [inline synapse/handlers/directory.py]: The `requester` is a separate type, which includes a user. Not sure why the functions in this module ask for both. I'm not even sure why the `requester` is needed.

**Pull request merged**